src/System.Text.Json/src/System/Text/Json/Linq/JTreeArray.cs

```diff
@@ -189,43 +189,62 @@ public JsonArray(IEnumerable<ulong> values) : this()
         }
 
         /// <summary>
-        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="decimal"/>s.
+        ///   Initializes a new instance of the <see cref="JTreeArray"/> class representing the specified collection of <see cref="decimal"/>s.
         /// </summary>
         /// <param name="values">Collection to represent.</param>
-        public JsonArray(IEnumerable<decimal> values) : this()
+        public JTreeArray(IEnumerable<decimal> values) : this()
         {
             foreach (decimal value in values)
             {
                 _list.Add(value);
             }
         }
 
+        /// <summary>
+        ///   Parses a string representing JSON document into a <see cref="JTreeArray"/>.
+        /// </summary>
+        /// <param name="json">JSON to parse.</param>
+        /// <param name="options">Options to control the parsing behavior.</param>
+        /// <returns><see cref="JTreeArray"/> representation of <paramref name="json"/>.</returns>
+        /// <exception cref="InvalidOperationException">
+        ///   Provided json text is not a JSON array.
+        /// </exception>
+        public static new JTreeArray Parse(string json, JTreeNodeOptions options = default)
+        {
+            JTreeNode node = JTreeNode.Parse(json, options);
+            if (node is JTreeArray JTreeArray)
+            {
+                return JTreeArray;
+            }
+            throw new InvalidOperationException(SR.Format(SR.JsonTypeMismatch, typeof(JTreeArray), node.GetType()));
+        }
+
         /// <summary>
         ///   Gets or sets the element at the specified index.
         /// </summary>
         /// <param name="idx">The zero-based index of the element to get or set.</param>
         /// <exception cref="ArgumentOutOfRangeException">
         ///   Index is less than 0.
         /// </exception>
-        /// <remarks>Null value is allowed and will be converted to the <see cref="JsonNull"/> instance.</remarks>
-        public JsonNode this[int idx]
+        /// <remarks>Null value is allowed and will be converted to the <see cref="JTreeNull"/> instance.</remarks>
+        public JTreeNode this[int idx]
         {
             get => _list[idx];
             set
             {
-                _list[idx] = value ?? new JsonNull();
+                _list[idx] = value ?? new JTreeNull();
```

Comment:
Wasn't there just a change to use a singleton of null?

Reply:
Yep, in the refactoring PR. Waiting for that one to get merged first, and then will update this PR. This PR won't make it, so I will port it manually to the other repo.